### PR TITLE
Main skeleton + CI

### PR DESCRIPTION
**Pull request by lhl2617**
Test plan:
```bash
$ go test ./...
?       github.com/SOMAS2020/SOMAS2020  [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/clients/team1   [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/clients/team2   [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/clients/team3   [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/clients/team4   [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/clients/team5   [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/clients/team6   [no test files]
?       github.com/SOMAS2020/SOMAS2020/internal/common  [no test files]
ok      github.com/SOMAS2020/SOMAS2020/internal/server  (cached)
```

```bash
$ go run .
2020/11/25 17:07:20 [1]: Echo: 'Hello World'
2020/11/25 17:07:20 [SERVER]: Received echo `Hello World` from client 1
2020/11/25 17:07:20 [2]: Echo: 'Hello World'
2020/11/25 17:07:20 [SERVER]: Received echo `Hello World` from client 2
2020/11/25 17:07:20 [3]: Echo: 'Hello World'
2020/11/25 17:07:20 [SERVER]: Received echo `Hello World` from client 3
2020/11/25 17:07:20 [4]: Echo: 'Hello World'
2020/11/25 17:07:20 [SERVER]: Received echo `Hello World` from client 4
2020/11/25 17:07:20 [5]: Echo: 'Hello World'
2020/11/25 17:07:20 [SERVER]: Received echo `Hello World` from client 5
2020/11/25 17:07:20 [6]: Echo: 'Hello World'
2020/11/25 17:07:20 [SERVER]: Received echo `Hello World` from client 6
```

CI initiated:
https://github.com/SOMAS2020/SOMAS2020/pull/2/checks?check_run_id=1454910250

image

image

### CODE REVIEW

**Comment by lhl2617** [comment]
I can't seem to get the protected_folder check working. Leaving that there for the moment. Will investigate in the future. Let's merge this first.

File: .github/workflows/protected_folder.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    branches: [ main ]
+    paths:
+    - 'internal/server/**'
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    - name: Warning
+      run: |
+        echo "::warning This diff contains changes in protected folders!"
```
Comment (lhl2617) [inline .github/workflows/protected_folder.yml]: IDK why this isn't working, so if someone volunteers to check, thanks!

File: .github/workflows/protected_folder.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+    - 'internal/server/**'
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    - name: Warning
```
Comment (preetl) [inline .github/workflows/protected_folder.yml]:
```suggestion
    runs-on: ubuntu-20.04
    steps:
    - name: Warning

```

**Comment by lhl2617** [comment]
thanks @preetl  and @QFSW  for fixing it! unfortunately it seems to be unable to show a protected folder write as a warning signal on the PR board, @QFSW  will be looking on having it be commented instead. 

Can I has stamp on this PR so we can land it pls? :D

**Review by QFSW — APPROVED** [review]

**Pull request merged**